runtime/queries/zig/highlights.scm

```diff
@@ -0,0 +1,250 @@
+[
+  (container_doc_comment)
+  (doc_comment)
+  (line_comment)
+] @comment
+
+;field in top level decl, and in struct, union...
+(ContainerField
+  (IDENTIFIER) @property
+  (SuffixExpr (IDENTIFIER) @type)?
+)
+
+; INFO: field become a function if type is a function?
+; const u = union { this_is_function: fn () void };
+(ContainerField
+  (IDENTIFIER) @function
+  (SuffixExpr (FnProto))
+)
+
+;enum and tag union field is constant
+(
+  [
+    ; union(Tag){}
+    (ContainerDeclType (SuffixExpr (IDENTIFIER) @type))
+
+    ; enum{}
+    (ContainerDeclType "enum")
+  ]
+  (ContainerField (IDENTIFIER) @constant)?
+)
+
+; INFO: .IDENTIFIER is a field?
+(SuffixExpr 
+  "."
+  (IDENTIFIER) @property
+)
+
+; error.OutOfMemory;
+(SuffixExpr 
+  "error"
+  "."
+  (IDENTIFIER) @constant
+)
+
+(VarDecl
+  (IDENTIFIER) @type
+  [
+    ; const IDENTIFIER = struct/enum/union...
+    (SuffixExpr (ContainerDecl))
+
+    ; const A = u8;
+    (SuffixExpr (BuildinTypeExpr))
+  ]
+)
+
+; const fn_no_comma = fn (i32, i32) void;
+(VarDecl
+  (IDENTIFIER) @function
+  (SuffixExpr (FnProto))
+)
+
+; var x: IDENTIFIER
+type: (SuffixExpr (IDENTIFIER) @type)
+
+; IDENTIFIER{}
+constructor: (SuffixExpr (IDENTIFIER) @constructor)
+
+;{.IDENTIFIER = 1}
+(FieldInit (IDENTIFIER) @property)
+
+; var.field
+(SuffixOp (IDENTIFIER) @property)
+
+; var.func().func().field
+( 
+  (SuffixOp
+    (IDENTIFIER) @function
+  )
+  .
+  (FnCallArguments)
+)
+; func()
+( 
+  (
+    (IDENTIFIER) @function
+  )
+  .
+  (FnCallArguments)
+)
+
+; functionn decl
+(FnProto
+  (IDENTIFIER) @function
+  (SuffixExpr (IDENTIFIER) @type)?
+  ("!")? @function.macro
+)
+
+(ParamDecl 
+  (ParamType (SuffixExpr (IDENTIFIER) @variable.parameter))
+)
+
+(ParamDecl 
+  (IDENTIFIER) @variable.parameter
+  ":"
+  [
+    (ParamType (SuffixExpr (IDENTIFIER) @type))
+    (ParamType)
+  ]
+)
+
+(SwitchItem 
+  (SuffixExpr
+    "."
+    .
+    (IDENTIFIER) @constant
+  )
+)
+
+(BUILTINIDENTIFIER) @function.builtin
```

Comment:
This should also be in the `@function.builtin` block.

Reply:
Done.